zivid_camera/src/zivid_camera.cpp

```diff
@@ -169,12 +173,24 @@ void runFunctionAndCatchExceptions(
   }
 }
 
-template<typename Logger>
+template <typename Function, typename Logger>
+auto runFunctionAndCatchExceptionsAndRethrow(Function && function, const Logger & logger)
+{
+  try {
+    return function();
+  } catch (const Zivid::Exception & exception) {
+    const auto exception_message = Zivid::toString(exception);
+    RCLCPP_ERROR_STREAM(
+      logger, "A function threw a Zivid::Exception: \"" + exception_message + "\"");
+    throw;
```

Comment:
let us just catch all `std::exception` here, there are some SDK functions that throw a non Zivid::Exception